azure/datalake/store/transfer.py

```diff
@@ -240,9 +245,14 @@ def __init__(self, adlfs, transfer, merge=None, nthreads=None,
         self._chunked = chunked
         self._unique_temporary = unique_temporary
         self._unique_str = uuid.uuid4().hex
+        self._progress_callback=progress_callback
+        self._progress_lock = threading.Lock()
```

Comment:
I am concerned that this will result in a big performance hit for folders containing lots of small items (such as a folder with 100,000 1MB files). I like the idea, but we will probably need logic in the CLI to either disable progress by default or be smart enough to disable it in situations with known performance implications.